onnxruntime/core/graph/graph.cc

```diff
@@ -4046,6 +4047,229 @@ Status Graph::AddConstantProtoAsInitializer(const ONNX_NAMESPACE::NodeProto& nod
   return Status::OK();
 }
 
+static void RenameSubgraphDependentNames(const InlinedHashMap<std::string, std::string>& name_mapping,
+                                         Graph& graph) {
+  for (auto& node : graph.Nodes()) {
+    if (node.ContainsSubgraph()) {
+      for (auto& [name, subgraph] : node.GetAttributeNameToMutableSubgraphMap()) {
+        RenameSubgraphDependentNames(name_mapping, *subgraph);
+      }
+    }
+
+    // NodeArgs need to be updated
+    auto input_defs = node.MutableInputDefs();
+    for (auto& input_def : node.MutableInputDefs()) {
+      auto hit = name_mapping.find(input_def->Name());
```

Comment:
Should if we check if the InputExists? Or, are you relying on the map not containing an entry for an empty string name?